core/src/main/java/greencity/controller/TipsAndTricksController.java

```diff
@@ -0,0 +1,149 @@
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @GetMapping("/{id}")
+    public ResponseEntity<TipsAndTricksDto> getTipsAndTricksById(@PathVariable Long id) {
+        return ResponseEntity.status(HttpStatus.OK).body(tipsAndTricksService.findDtoById(id));
+    }
+
+    /**
+     * Method for getting all tips & tricks by page.
+     *
+     * @return PageableDto of {@link TipsAndTricksDto} instances.
+     */
+    @ApiOperation(value = "Find all tips & tricks by page.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @GetMapping("")
+    @ApiPageable
+    public ResponseEntity<PageableDto<TipsAndTricksDto>> findAll(@ApiIgnore Pageable page) {
+        return ResponseEntity.status(HttpStatus.OK).body(tipsAndTricksService.findAll(page));
+    }
+
+    /**
+     * Method for deleting {@link TipsAndTricks} by its id.
+     *
+     * @param tipsAndTricksId {@link TipsAndTricks} id which will be deleted.
+     * @return id of deleted {@link TipsAndTricks}.
+     */
+    @ApiOperation(value = "Delete tips & tricks.")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 303, message = HttpStatuses.SEE_OTHER),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @DeleteMapping("/{tipsAndTricksId}")
+    public ResponseEntity<Object> delete(@PathVariable Long tipsAndTricksId) {
+        tipsAndTricksService.delete(tipsAndTricksId);
+        return ResponseEntity.status(HttpStatus.OK).build();
+    }
+
+    /**
+     * Method for getting all tips & tricks by tags.
+     *
+     * @return list of {@link TipsAndTricksDto} instances.
+     */
+    @ApiOperation(value = "Get tips & tricks by tags")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN),
+    })
+    @GetMapping("/tags")
+    @ApiPageable
+    public ResponseEntity<PageableDto<TipsAndTricksDto>> getTipsAndTricks(
+        @ApiIgnore Pageable page,
+        @ApiParam(value = "Tags to filter (if do not input tags get all)")
+        @RequestParam(required = false) List<String> tags
+    ) {
+        if (tags == null || tags.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.OK).body(
+                tipsAndTricksService.findAll(page));
+        }
+        return ResponseEntity.status(HttpStatus.OK).body(tipsAndTricksService.find(page, tags));
```

Comment:
Move logic with if to service. Services are responsible for business logic

Reply:
Moved this logic to the TipsAndTricksService, method 'find'